DOCUMENTATION.md

```diff
@@ -0,0 +1,305 @@
+
+**Example:**
+
+```CSS
+.present .rotating {
+  transform: rotate(-10deg);
+  transition-delay: 0.25s;
+}
+```
+
+The `.past` class is added to all [Step Elements](#step-element) that have been visited at least once.
+
+**Example:**
+
+```CSS
+.past {
+  display: none;
+}
+```
+
+### Current Active Step (.active class)
+
+The `.active` class is added to the [Step Element](#step-element) that is currently visible at the center of the camera.
+
+**Example:**
+
+```CSS
+.step {
+  opacity: 0.3;
+  transition: opacity 1s;
+}
+.step.active {
+  opacity: 1
+}
+```
+
+At the same time, the `impress-on-*` class is added to the body element, the class name represents the active [Step Element](#step-element) id. This allows for custom global styling, since you can't match a CSS class backwards from the active [Step Element](#step-element) to the `body`.
+
+**Example:**
+
+```CSS
+.impress-on-overview .step {
+    opacity: 1;
+    cursor: pointer;
+}
+.impress-on-step-1,
+.impress-on-step-2,
+.impress-on-step-3 {
+  background: LightBlue;
+}
+```
+
+### Progressive Enhancement (.impress-not-supported class)
+
+The `.impress-not-supported` class is added to the `body` element if the browser doesn't support the features required by impress.js to work, it is useful to apply some fallback styles in the CSS.
+
+It's not necessary to add it manually on the `body` element. If the script detects that the browser lacks important features it will add this class.
+
+It is recommended to add the class manually to the `body` element though, because that means that users without JavaScript will also get fallback styles. When impress.js script detects that the browser supports all required features, the `.impress-not-support` class will be removed from the `body` element.
+
+**Example:**
+
+```CSS
+.impress-not-supported .step {
+  display: inline-block;
+}
+```
+
+## JavaScript
+
+### impress().init()
+
+Initializes impress.js globally in the page. Only one instance of impress.js is supported per document.
+
+**Example:**
+
+```JavaScript
+impress().init();
+```
+
+Triggers the `impress:init` event in the [Root Element](#root-element) after the presentation is initialized.
+
+**Example:**
+
+```JavaScript
+var rootElement = document.getElementById( "impress" );
+rootElement.addEventListener( "impress:init", function() {
+  console.log( "Impress init" );
+});
+impress().init();
+```
+
+### impress().next()
+
+Navigates to the next step of the presentation using the [`goto()` function](#impressgotostepindexstepelementidstepelement-duration).
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.next();
+```
+
+### impress().prev()
+
+Navigates to the previous step of the presentation using the [`goto()` function](#impressgotostepindexstepelementidstepelement-duration).
+
+**Example:**
+
+```JavaScript
+var api = impress();
+api.init();
+api.prev();
+```
+
+### impress().goto(stepIndex|stepElementId|stepElement, [duration])
```

Comment:
I would add spaces `goto( stepIndex | stepElementId | stepElement, [duration] )` to make it more readable.